sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/json/JacksonGenerator.scala

```diff
@@ -202,5 +202,9 @@ private[sql] class JacksonGenerator(
    */
   def write(array: ArrayData): Unit = writeArray(writeArrayData(array, arrElementWriter))
 
+  def write(map: MapData, mapType: MapType): Unit = {
+    writeObject(writeMapData(map, mapType, makeWriter(mapType.valueType)))
```

Comment:
@goldmedal Similar to `arrElementWriter`, let's create a `mapValueWriter` at the beginning of `JacksonGenerator`?